plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,434 @@
+					sentinelTags["sentinel_ip"] = sm["ip"]
+					sentinelTags["sentinel_port"] = sm["port"]
+					sentinelTags["master_name"] = m["name"]
+					sentinelFields := make(map[string]interface{})
+
+					for key, val := range sm {
+						if ival, err := strconv.ParseInt(val, 10, 64); err == nil {
+							sentinelFields[key] = ival
+							continue
+						}
+
+						if fval, err := strconv.ParseFloat(val, 64); err == nil {
+							sentinelFields[key] = fval
+							continue
+						}
+
+						sentinelFields[key] = val
+					}
+
+					acc.AddFields("redis_sentinels", sentinelFields, sentinelTags)
+				}
+
+				// ------------------------------------------------------------
+
+				// Check other Replicas
+				replicaTags := client.tags
+
+				// Use client.Sentinel.Slaves() for >6.15.3 of go-redis
+				replicasCmd := redis.NewSliceCmd("sentinel", "replicas", m["name"])
+				if srErr := client.sentinel.Process(replicasCmd); srErr != nil {
+					lastError.Add(srErr)
+					return
+				}
+
+				replicas, replicasErr := replicasCmd.Result()
+				if replicasErr != nil {
+					lastError.Add(replicasErr)
+					return
+				}
+
+				for _, replica := range replicas {
+					rm := toMap(replica.([]interface{}))
+					replicaTags["replica_ip"] = rm["ip"]
+					replicaTags["replica_port"] = rm["port"]
+					replicaTags["master_name"] = m["name"]
+					replicaFields := make(map[string]interface{})
+
+					for key, val := range rm {
+						if ival, err := strconv.ParseInt(val, 10, 64); err == nil {
+							replicaFields[key] = ival
+							continue
+						}
+
+						if fval, err := strconv.ParseFloat(val, 64); err == nil {
+							replicaFields[key] = fval
+							continue
+						}
+
+						replicaFields[key] = val
+					}
+
+					acc.AddFields("redis_replicas", replicaFields, replicaTags)
+				}
+
+				// ------------------------------------------------------------
+
+				// Use client.Sentinel.Info() for >6.15.3 of go-redis
+				infoCmd := redis.NewStringCmd("info")
+				if iErr := client.sentinel.Process(infoCmd); iErr != nil {
+					lastError.Add(iErr)
+					return
+				}
+
+				info, infoErr := infoCmd.Result()
+				if infoErr != nil {
+					lastError.Add(infoErr)
+					return
+				}
+
+				rdr := strings.NewReader(info)
+				gatherSentinelInfoOutput(rdr, acc, client.BaseTags())
+			}
+		}(client, acc)
+	}
+
+	wg.Wait()
+
+	return lastError.Get()
+}
+
+func init() {
+	inputs.Add("redis_sentinel", func() telegraf.Input {
+		return &RedisSentinel{}
+	})
+}
+
+// gatherSentinelInfoOutput parses `INFO` command output
+// Largely copied from the Redis input plugin's gatherInfoOutput()
+func gatherSentinelInfoOutput(
+	rdr io.Reader,
+	acc telegraf.Accumulator,
+	tags map[string]string,
+) {
+	scanner := bufio.NewScanner(rdr)
+	fields := make(map[string]interface{})
+
+	var section string
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) == 0 {
+			continue
+		}
+
+		if line[0] == '#' {
+			if len(line) > 2 {
+				section = line[2:]
```

Comment:
I'm slightly confused why this is set here, but the loop continues, can you add comments as to why this is needed. (It seems to be tied to line 396, and I imagine the output is something like the following where section is the section heading?):
```
Server
key:value
```